src/org/usfirst/frc4904/autonly/Field.java

```diff
@@ -29,5 +87,36 @@ public Piece(Alliance team, char reference) {
 			right = team;
 			left = Alliance.values()[1 - team.ordinal()];
 		}
+		
+		/** Constructs the piece based on the actual left and right alliances.
+		 * 
+		 * @param left The alliance on the left side of this piece.
+		 * @param right The alliance on the right side of this piece.
+		 */
+		public Piece(Alliance left, Alliance right) { 
+			this.left = left;
+			this.right = right;
+		}
+		
+		@Override
+		public String toString() {
+			return "Piece{L:'" + left.name() + "' R:'"+ right.name() +"'}";
+		}
+		
+		/** Tells us if the left side of the pieces is ours.
+		 * 
+		 * @return true if the left side is ours, false if it is not.
+		 */
+		public boolean isLeftOurs(){
+			return (DriverStation.getInstance().getAlliance() == this.left) ? true : false;
```

Comment:
`isLeftOurs()` and `isRightOurs()` should be changed to ensure that it's not comparing two Alliance.Invalid values. We don't want to run any side-based auton if the field didn't send us the proper field configuration value.